### PR TITLE
CLI: Avoid id changes after `storiesof-to-csf` migration

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: sometimes running `storiesof-to-csf` codemod leads to story id changes

## What I did
1. Replaced camelCase with PascalCase in export names which rules out clashes with reserved keywords
2. Replaced `Story` suffix with `_` prefix which gets stripped during conversion to id

### CODE REVIEW

File: lib/codemod/src/lib/utils.js [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-
-export const isReserved = name => RESERVED.exec(name);
+import upperFirst from 'lodash/upperFirst';
 
 export const sanitizeName = name => {
-  let key = camelCase(name);
-  if (isReserved(key)) {
-    key = `${key}Story`;
-  }
+  let key = upperFirst(camelCase(name));
```
Comment (Hypnosphi) [inline lib/codemod/src/lib/utils.js]: I wanted to use pipeline operator here, but I'm not sure whether it's OK with other contributors:

```js
let key = name |> camelCase |> upperFirst
```

Comment (shilman) [reply lib/codemod/src/lib/utils.js]: 🤔 stage 1 ....

**Review by shilman — APPROVED** [review]
This makes sense & is definitely an improvement over the existing behavior.

My one concern is about PascalCase, since I don't know whether we should encourage that convention for story exports. What do you think about changing it to `_${camelCase(name)}` in those cases?

Curious to hear what you and others think.

**Comment by atanasster** [comment]
Just my 0.02c camelCase seems more natural in javascript projects /coming from a 20+ years Pascal guy :)/

**Comment by tmeasday** [comment]
I discussed this a little with @shilman and I think I agree that the "prefix reserved words with _" solution is probably a better way to achieve the same goal. Here's my thought process:

1. If we update the codemod to PascalCase but not our examples, people using the codemod are going to be put off: "this doesn't look like the docs, did it break or something?". I don't love that.
2. If we update our examples, the first thing a new user to storybook is going to see is something like:

```js
export Welcome = () => <Welcome />;
```

I guess technically there's nothing wrong with a pascal-cased export but I reckon if I was a new user it would put me off.

**Comment by Hypnosphi** [comment]
> camelCase seems more natural in javascript projects

But using PascalCase for components in JSX-based frameworks and libraries like React is quite a common practice. And the stories look pretty much like components (even if technically they aren't), and we even allow React hooks in them. So maybe PascalCasing them by default is a good thing? Among other things, it would:
1. Remove the need to suppress `react-hooks/rules-of-hooks` ESLint rule when using React hooks. Right now it complains because it doesn't treat a camelCased function as a component
2. Make it easier to reuse stories inside other stories. You could do just `<MyStory />` without reassignments like `const MyStory = myStory`

> If we update the codemod to PascalCase but not our examples, people using the codemod are going to be put off: "this doesn't look like the docs, did it break or something?". I don't love that.

It makes sense to me to update the examples as well. I can do it on approval

> If we update our examples, the first thing a new user to storybook is going to see is something like:
> ```js
> export Welcome = () => <Welcome />;
> ```

Not really. What they actually would see in case of official-storybook is
```js
export const ToStorybook = () => <Welcome showApp={linkTo('Button')} />;
```

**Comment by tmeasday** [comment]
That's a good point @Hypnosphi -- I think I am convinced, although it seems an important change.

**Comment by shilman** [comment]
That might be the case for React, but not for any of our other frameworks where we're returning JS objects... My gut still says `_camelCase` is the way to go. That said, I'm not married to that, so if you feel strongly about `PascalCase` then I reluctantly approve. 😄

**Comment by Hypnosphi** [comment]
OK let me migrate examples and CLI templates then

File: lib/codemod/src/lib/utils.js (hunk 1 above)
Comment (ndelangen) [reply lib/codemod/src/lib/utils.js]: Let's not use language features that typescript doesn't support?

**Pull request merged**